refactor ref_data to use RefCounted

eh-jogos opened the pull request:
This is a "proposal" for a way to use ref_data with RefCounted objects instead of Dictionaries. 

I think it will be a little bit easier to create new ones when you don't want to override a lot of stats, adds type completion when using it in code, and automates passing the data from UnitData to ActorStats, so we don't have to give it maintenance, just add the same property on both and it works.

Snayff commented:
I've looked at this twice now and it isnt clicking for me. I'll pull the branch and play around locally, see if that does the trick.

eh-jogos committed:
refactor ref_data to use RefCounted

eh-jogos committed:
Fix _create_actor_stats

The "automation" I tried to do doesn't work because almost no property
in ActorStats has the same name as in UnitData

eh-jogos commented:
Just updated this PR rebasing it to curent develop and fixing some mistakes
